doc/help/elektra-highlevel-gen.md

```diff
@@ -0,0 +1,415 @@
+# elektra-highlevel-gen(7) -- High-level API code-generation advanced features
+
+This document focuses on the advanced features of the high-level API code-generator template. We assume you already familiarized yourself
+with the basic features explained in [`kdb-gen-highlevel(1)`](kdb-gen-highlevel.md).
+
+## Configuration Options
+
+The parameters that are relevant to the concepts described here are (for the rest see [`kdb-gen-highlevel(1)](kdb-gen-highlevel.md)):
+
+- `enumConv`: allowed values: `strcmp`, `trie`, `default` (default)
```

Comment:
And the enumConf option is not needed if the default works properly. Did you benchmark if there are really scenarios where the trie is noticeable slower?

Reply:
> the enumConf option is not needed if the default works properly

The option is already there and it has basically no impact on the code (of the code-generator template). It also doesn't complicate the documentation, since even without the option, we would need to explain the two different outcomes of using enums. So I don't really see any reason to remove the option.

> Did you benchmark if there are really scenarios where the trie is noticeable slower?

I only did very quick tests. As far as I can tell, the trie implementation is quicker even if the enum values match the regexes `a{31}a` and `a{31}b`. But the runtime of the trie version seems to increase quicker than the strcmp version (with respect to the string length).

However, we are talking about a difference of a few hundred milliseconds per 10.000.000 executions, so it hardly matters. (Also my CPU is quite old, newer CPUs might perform better in strcmp.)

The main reason why I didn't use the trie version all the time is, that is becomes unreadable very very quickly. That is also why the option exists. If you want optimal performance `trie` *might* be better, for readability `strcmp` is probably better.

Reply:
> we would need to explain the two different outcomes of using enums.

If we do not have the option we would not need to explain anything. But you already gave a good reason why we need the option.

> The main reason why I didn't use the trie version all the time is, that is becomes unreadable very very quickly. That is also why the option exists.

Then say exactly this. Why bringing up wild speculations of performance if this option is actually about being able to understand the output of the generator. (Which is a perfectly good reason to have the option.)